File: 121-Best-Time-to-Buy-and-Sell-Stock.py
class Solution(object):
    def maxProfit(self, prices):
        l, r = 0, 1
        max_profit = 0

        while r < len(prices):
            if prices[l] < prices[r]:
                profit = prices[r] - prices[l]
                max_profit = max(max_profit, profit)
            else:
                l = r
            r += 1

        return max_profit

File: neetcode/121-Best-Time-to-Buy-and-Sell-Stock.ts
function maxProfit(prices: number[]): number {
  let l = 0, r = 1
  let max = 0

  while (r < prices.length) {
    let cur = prices[r] - prices[l]
    max = Math.max(cur, max)

    if (prices[r] >= prices[l]) r++
    else l++
  }

  return max
};